promql/promqltest/testdata/functions.test

```diff
@@ -1022,6 +1022,13 @@ eval instant at 1m last_over_time(data[2m])
 	data{type="some_nan3"} 1
 	data{type="only_nan"} NaN
 
+eval instant at 1m avg_over_time(data[2m])
+	data{type="some_nan3"} 0.5
```

Comment:
DEMONSTRATES BUG related to original issue: [this test failure](https://github.com/prometheus/prometheus/actions/runs/12008490726/job/33471236452?pr=15451#step:4:17362) proves that `avg_over_time` over [[NaN, 0, 1]](https://github.com/prometheus/prometheus/pull/15451/files#diff-f277c3549a17549dbc047c4f13cdeebd450a0a69b2a568f7dde50935583b3540R1001) produces `NaN` according to former (and still current) behavior instead of 0.5 expected on this line